connectors/utils.py

```diff
@@ -401,3 +401,14 @@ async def func_to_execute(*args, **kwargs):
 
 def e2str(entry):
     return entry.name.lower()
+
+
+def str2e(name, enum_class):
+    if name is None:
```

Comment:
it seems that this function will return `None` when the name is not know. Shouldn't we raise an error instead? I can think of problems where None will be used silently and can't think of scenarios where we ask for a name that is not supposed to exist in the Enum

Reply:
Gotcha. I guess if we use strings in the enums we can have an explicit set:

```
enum_value = None if value is None else JobStatus(value)
```

Reply:
or an unset value:

```
>>> class MyEnum(Enum):
...     one = 'one'
...     two = 'two'
...     unset = None
...
>>> MyEnum(None)
<MyEnum.unset: None>
>>> MyEnum('one')
<MyEnum.one: 'one'>
```